G/gmsh/build_tarballs.jl

```diff
@@ -39,13 +44,45 @@ platforms = expand_cxxstring_abis(supported_platforms())
 # The products that we will ensure are always built
 products = [
     LibraryProduct(["libgmsh", "gmsh"], :libgmsh),
-    FileProduct("lib/gmsh.jl",:gmsh_api)
+    FileProduct("lib/gmsh.jl", :gmsh_api)
+]
+
+# Some dependencies are needed or available only on certain platforms
+x11_platforms = filter(p ->Sys.islinux(p) || Sys.isfreebsd(p), platforms)
+hdf5_platforms = [
+    Platform("x86_64", "linux"),
+    Platform("aarch64", "linux"; libc="glibc"),
+    Platform("x86_64", "macos"),
+    Platform("aarch64", "macos"),
+    Platform("x86_64", "windows"),
+    Platform("i686", "windows")
 ]
 
 # Dependencies that must be installed before this package can be built
 dependencies = [
-    Dependency(PackageSpec(name="CompilerSupportLibraries_jll", uuid="e66e0078-7015-5450-92f7-15fbd957f2ae")),
+    BuildDependency("Xorg_xorgproto_jll"; platforms=x11_platforms),
+    Dependency("Cairo_jll"),
+    Dependency("CompilerSupportLibraries_jll"; platforms=filter(!Sys.isbsd, platforms)),
+    Dependency("FLTK_jll"),
+    Dependency("FreeType2_jll"),
+    Dependency("GLU_jll"; platforms=x11_platforms),
+    Dependency("GMP_jll"; compat="6.2"),
+    Dependency("HDF5_jll"; platforms=hdf5_platforms, compat="1.12"),
```

Comment:
Our build of HDF5 1.12.0 [doesn't have aarch64-apple-darwin](https://github.com/JuliaBinaryWrappers/HDF5_jll.jl/releases/tag/HDF5-v1.12.0%2B1), contrary to what you're saying above in `hdf5_platforms`.  You have to either use [1.12.1](https://github.com/JuliaBinaryWrappers/HDF5_jll.jl/releases/tag/HDF5-v1.12.1%2B0) (recommended) or remove aarch64-apple-darwin from the list above.

Reply:
Resolved by using 1.12.1 which supplies the aarch64 build for MacOS.